framework/test/gtest/BundleContextTest.cpp

```diff
@@ -192,4 +192,115 @@ TEST(BundleContextTest, NoSegfaultWithServiceFactory)
     }
     thread.join();
 }
+
+template <typename ServiceT>
+bool
+verifyOrdering(std::vector<cppmicroservices::ServiceReference<ServiceT>>& refs)
+{
+    if (refs.size() > 0)
+    {
+        auto last = refs[0];
+        for (auto const& ref : refs)
+        {
+            if (!(any_cast<int>(ref.GetProperty(Constants::SERVICE_RANKING))
+                  <= any_cast<int>(last.GetProperty(Constants::SERVICE_RANKING))))
+            {
+                return false;
+            }
+
+            last = ref;
+        }
+    }
+    return true;
+}
+
+TEST(BundleContextTest, TestGetServiceReferenceOrderingBasic)
```

Comment:
this test should be written using [value parameterized tests](https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests)

Reply:
fixed